src/language_features/completion.rs

```diff
@@ -196,21 +192,23 @@ pub fn editor_completion(
             // we'll need to perform some transformations on the completion commands.
             if ctx.config.snippet_support && x.insert_text_format == Some(InsertTextFormat::SNIPPET)
             {
+                lazy_static::lazy_static! {
+                    static ref SNIPPET_TABSTOP_RE: Regex = Regex::new(r"\$(?P<i>\d+)").unwrap();
+                    static ref SNIPPET_PLACEHOLDER_RE: Regex = Regex::new(r"\$\{(?P<i>\d+):?(?P<placeholder>[^}]+)\}").unwrap();
```

Comment:
If we're not using the named captured group "i", then we should remove it to reduce confusion.

I think one usecase of snippets that don't just use lexical order for placeholders is something like this:

	print(r"${1:foo}")
	print(${1:foo})

This means that the first placeholder should get two cursors. In this case this allows to print a variable and its value. VSCode supports this.
Not sure if any language server ever sends something like this. Anyay, not important now.

Reply:
I really want to support this yes, it would be a part of rewriting select-next-placeholders. And yes, thats why the capture group is named

Reply:
Sure. First we should find a language server that sends snippets with either multiple cursors or with indices in non-lexical order.

Reply:
I would assume rust-analyzer does, but to be fair, I haven't checked. I'll look at that if/when I continue down this road